pkg/arvo/app/ping.hoon

```diff
@@ -20,284 +20,42 @@
 ::
 ++  nat-timeout  ~s25
 ::
-::  How often to check our IP when we know we're not behind a NAT.
-::
-++  ip-timeout   ~m5
-::
-::  Chosen because it's run by Cloudflare, and others I tried were
-::  inconsistently slow.
-::
-++  ip-reflector  'https://icanhazip.com'
-::
 +$  card  card:agent:gall
-+$  ship-state
-  $%  [%idle ~]
-      [%poking ~]
-      [%http until=@da]
-      [%waiting until=@da]
-  ==
-+$  state-2
-  $:  %2
-      ships=(set ship)
-      nonce=@ud
-      $=  plan
-      $~  [%nat ~]
-      $%  [%nat ~]
-          [%pub ip=(unit @t)]
-          [%off ~]
-          [%one ~]
-      ==
+::
++$  state-3
+  $:  %3
+     mode=?(%formal %informal)
+     pokes=@ud
+     timer=(unit [=wire date=@da])
```

Comment:
This is not enough information to cancel timers if we decide to do so later. The duct on which we set the timer will be differen ift `!=(our src):bowl`. If we care about getting around this, we need to use the "self-poke trick" to get a consistent duct before we actually set the timer. I don't think we need to care though.